src/modules/sync/components/sync-sites-modal-selector.tsx

```diff
@@ -286,9 +290,12 @@ function SiteItem( {
 				<div className="flex gap-2">
 					{ ! isPressable && (
 						<>
-							<EnvironmentBadge type="production" selected={ isSelected } />
-							{ site.stagingSiteIds.length > 0 && (
-								<EnvironmentBadge type="staging" selected={ isSelected } />
+							<EnvironmentBadge type={ getSiteEnvironment( site ) } selected={ isSelected } />
+							{ stagingSite && (
+								<EnvironmentBadge
+									type={ getSiteEnvironment( stagingSite ) }
+									selected={ isSelected }
+								/>
```

Comment:
I see the older code can work too, and it would remove the need of passing `allSites`. However, the benefit of the new approach is that we centralize all the logic of identifying the environment inside a single function `getSiteEnvironment`.

Reply:
Good catch! At the end I decided to restore the original code as I think it is more readable and simple: https://github.com/Automattic/studio/pull/1645/commits/dd7577d1d997d0db88bb909c741ec17a25788259.